aep/general/0162/aep.md.j2

```diff
@@ -1,5 +1,396 @@
+      "post": {
+        "requestBody": {
+          "content": {
+            "application/json": {
+              "schema": {
+                "$ref": "#/components/schemas/RollbackBookRequest"
+              }
+            }
+          }
+        },
+        "responses": {
+          "200": {
+            "description": "Successful rollback",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/BookRevision"
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  },
+  "components": {
+    "schemas": {
+      "RollbackBookRequest": {
+        "type": "object",
+        "properties": {
+          "path": {
+            "type": "string"
+          }
+        },
+        "required": ["path"]
+      }
+    }
+  }
+}
+```
+
+{% endtabs %}
+
+- The method **must** use the `POST` HTTP verb.
+- The method **should** return a resource revision.
+- The request message **must** have a `path` field, referring to the resource
+  revision whose configuration the resource should be rolled back to.
+  - The field **must** be [annotated as required][aip-203].
+  - The field **must** identify the [resource type][aip-123] that it
+    references.
+
+### Child resources
+
+Resources with a revision history **may** have child resources. If they do,
+there are two potential variants:
+
+- Child resources where each child resource is a child of the parent resource
+  as a whole.
+- Child resources where each child resource is a child of _a single revision
+  of_ the parent resource.
+
+APIs **should not** include multiple levels of resources with revisions, as
+this quickly becomes difficult to reason about.
+
+### Standard methods
+
+Any standard methods **must** implement the corresponding AIPs (AIP-131,
+AIP-132, AIP-133, AIP-134, AIP-135), with the following additional behaviors:
+
+- List methods: By default, revisions in the list response **must** be ordered
+  in reverse chronological order. User can supply `order_by` to override the
+  default behavior.
+- If the revision supports aliasing, a delete method with the resource name
+  of the alias (e.g. `revisions/1.0.2`) **must** remove the alias instead of
+  deleting the resource.
+
+As revisions are nested under the resource, also see [cascading delete][].
+
+## Rationale
+
+### Abstract revisions as nested collection
+
+Revisions being resources under nested collection make revisions a first class
+citizen.
+
+- Revisions can offer standard get, list, and delete methods.
+- It retains the flexibility of extending new fields to a revision in addition to
+  the resource message.
+
+### Output only resource configuration
+
+Although it was an option to have the revision take in the resource
+configuration as part of the create method, doing so would have allowed users to
+submit resource configuration for a revision that the resource was never in.
```

Comment:
basically - the snapshot field is read-only because if it was mutable, someone could create a snapshot with values that the resource never had - a fake snapshot, basically.

Reply:
Ah, I understand now.  Some suggestions that would help make that clearer:

* Fix the hyphen placement in the section header ("Output-only resource..." rather than "Output only-resource...")
* Remove replace "resource configuration" (which is not a defined term AFAIK) with something else, maybe "snapshots"
* Break it up into two sentences and kill the trailing preposition (I'm not puritanical about trailing prepositions, but in this case I think it does contribute to the sentence being hard to parse).

That said...do we need to explain this guidance in particular?  It seems pretty obvious to me why the snapshot field would be output-only.

Reply:
I'm fine with removing it!